istioctl/cmd/istioctl/main.go

```diff
@@ -300,15 +301,26 @@ istioctl get virtualservice bookinfo
 				return err
 			}
 
+			getByName := len(args) > 1
+			if getAllNamespaces && getByName {
+				return errors.New("a resource cannot be retrieved by name across all namespaces")
+			}
+
+			var ns string
+			if getAllNamespaces {
+				ns = v1.NamespaceAll
+			} else {
+				ns, _ = handleNamespaces(v1.NamespaceAll)
```

Comment:
Thanks @ayj , but here should be follows, otherwise, we will always return resources in all namespaces.

```
ns, _ = handleNamespaces(namespace)
```

Reply:
That wasn't my observation during testing. In practice, either seems to work the same way because either `namespace` is not-empty (line 828) and we use that or we select the defaultNamespace from user's config context.